generate/main.go

```diff
@@ -0,0 +1,180 @@
+		"DatabaseConnections":       "The number of database connections in use. Units: Count",
+		"DiskQueueDepth":            "The number of outstanding IOs (read/write requests) waiting to access the disk. Units: Count",
+		"FreeableMemory":            "The amount of available random access memory. Units: Bytes",
+		"FreeStorageSpace":          "The amount of available storage space. Units: Bytes",
+		"MaximumUsedTransactionIDs": "The maximum transaction ID that has been used. Applies to PostgreSQL. Units: Count",
+		"NetworkReceiveThroughput":  "The incoming (Receive) network traffic on the DB instance, including both customer database traffic and Amazon RDS traffic used for monitoring and replication. Units: Bytes/second",
+		"NetworkTransmitThroughput": "The outgoing (Transmit) network traffic on the DB instance, including both customer database traffic and Amazon RDS traffic used for monitoring and replication. Units: Bytes/second",
+		"OldestReplicationSlotLag":  "The lagging size of the replica lagging the most in terms of WAL data received. Applies to PostgreSQL. Units: Megabytes",
+		"ReadIOPS":                  "The average number of disk I/O operations per second. Units: Count/Second",
+		"ReadLatency":               "The average amount of time taken per disk I/O operation. Units: Seconds",
+		"ReadThroughput":            "The average number of bytes read from disk per second. Units: Bytes/Second",
+		"ReplicaLag":                "The amount of time a Read Replica DB instance lags behind the source DB instance. Applies to MySQL, MariaDB, and PostgreSQL Read Replicas. Units: Seconds",
+		"ReplicationSlotDiskUsage":  "The disk space used by replication slot files. Applies to PostgreSQL. Units: Megabytes",
+		"SwapUsage":                 "The amount of swap space used on the DB instance. Units: Bytes",
+		"TransactionLogsDiskUsage":  "The disk space used by transaction logs. Applies to PostgreSQL. Units: Megabytes",
+		"TransactionLogsGeneration": "The size of transaction logs generated per second. Applies to PostgreSQL. Units: Megabytes/second",
+		"WriteIOPS":                 "The average number of disk I/O operations per second. Units: Count/Second",
+		"WriteLatency":              "The average amount of time taken per disk I/O operation. Units: Seconds",
+		"WriteThroughput":           "The average number of bytes written to disk per second. Units: Bytes/Second",
+	}
+)
+
+func main() {
+	f, err := os.Create("metrics.go")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	packageTemplate.Execute(f, struct {
+		Timestamp time.Time
+		Metrics   []Metric
+	}{
+		Timestamp: time.Now(),
+		Metrics:   metrics,
+	})
+}
+
+var packageTemplate = template.Must(template.New("").Parse(`// Code generated by go generate; DO NOT EDIT.
+// This file was generated by robots at
+// {{ .Timestamp }}
+package main
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var Metrics = []Metric{
+{{- range .Metrics }}
```

Comment:
In templates, unlike in Go, iteration over a map is ordered, so we can drop metrics slice.

Reply:
I'm not sure what you mean here, so feel free to make PR that will explain it better.
In any case this is just generator, not used on runtime, but ok ;)